### PR TITLE
Convert [HandleProto] to [Plan.HandleConnection].

### PR DESCRIPTION
**Pull request by bgogul**

### CODE REVIEW

File: java/arcs/core/data/proto/HandleProtoDecoder.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Converts a [HandleProto] into [HandleConnection].
+ *
+ * If a type is not set in the [HandleProto], it is initialized to a newly created TypeVariable.
+*/
+fun HandleProto.decodeAsHandleConnection(mode: HandleMode) = HandleConnection(
+    // TODO(bgogul): name, fate, associatedHandles, ttl,
+    storageKey = StorageKeyParser.parse(storageKey),
+    mode = mode,
+    type = if (hasType()) type.decode() else TypeVariable("~$name"),
```
Comment (piotrswigon) [inline java/arcs/core/data/proto/HandleProtoDecoder.kt]: FWIW in TypeScript you would just say TypeVariable("name") and the `~` sign gets added during toString().

Comment (bgogul) [reply java/arcs/core/data/proto/HandleProtoDecoder.kt]: Removed the `~` now.

**Comment by bgogul** [comment]
> I didn't realize we have Proto stuff in core. It should really be in java/arcs/jvm instead - unless we have cross-platform proto stuff working?

Right now, I have a separate target https://github.com/PolymerLabs/arcs/blob/c54ad27ef24c270a93e7c05678a34277fdb5f70b/java/arcs/core/data/proto/BUILD#L17 that can be imported into an apk. This target uses `android_proto_library`. 

 I will be happy to move this to jvm if that makes more sense.

**Comment by piotrswigon** [comment]
@jasonwyatt what about protos is platform dependent? They seem to me as platform independent as stuff gets, WDYT?

**Comment by bgogul** [comment]
> @jasonwyatt what about protos is platform dependent? They seem to me as platform independent as stuff gets, WDYT?

@jasonwyatt, correct me if I am wrong. I think the issue is that the full protobuf library cannot be linked into an APK. The current solution with linking the protobuf lite library should work.

**Pull request merged**